VerbrauchZeitspanne/module.php

```diff
@@ -91,15 +120,74 @@ public function RequestAction($Ident, $Value)
+            } else {
+                if ($levelOfDetail == LOD_TIME) {
+                    $startDate = strtotime(date('H:i:s', $startDate), $this->getTime());
+                    $endDate = strtotime(date('H:i:s', $endDate), $this->getTime());
+                }
+
+                //FirstMinutes
+                $firstMinutesStart = strtotime(date('H:i', $startDate) . ':00', $startDate);
+                $this->SendDebug('FirstMinutsStart', date('H:i:s', $firstMinutesStart), 0);
+                $firstMinutesEnd = strtotime(date('H', $startDate) . ':59:59', $startDate);
+                $this->SendDebug('FirstMinutsEnd', date('H:i:s', $firstMinutesEnd), 0);
+                $firstMinutes = AC_GetAggregatedValues($acID, $variableID, 6 /* Minutes */, $firstMinutesStart, $firstMinutesEnd, 0);
+
+                //LastMinutes
+                $lastMinutesStart = strtotime(date('H', $endDate) . ':00:00', $endDate);
+                $this->SendDebug('LastMinutsStart', date('H:i:s', $lastMinutesStart), 0);
+                $lastMinutesEnd = strtotime(date('H:i', $endDate) . ':00', $endDate);
+                $this->SendDebug('LastMinutsEnd', date('H:i:s', $lastMinutesEnd), 0);
+                $lastMinutes = AC_GetAggregatedValues($acID, $variableID, 6 /* Minutes */, $lastMinutesStart, $lastMinutesEnd, 0);
+
+                if ($levelOfDetail == LOD_DATETIME) {
+                    //FirstHours
+                    $firstHoursStart = strtotime(date('H:00:00', $startDate) . ' next hour', $startDate);
+                    $this->SendDebug('FirstHoursStart', date('d.m.Y H:i:s', $firstHoursStart), 0);
+                    $firstHoursEnd = strtotime('23:59:59', $startDate);
+                    $this->SendDebug('FirstHoursEnd', date('d.m.Y H:i:s', $firstHoursEnd), 0);
+                    $firstHours = AC_GetAggregatedValues($acID, $variableID, 0 /* Hour */, $firstHoursStart, $firstHoursEnd, 0);
+
+                    //Days
+                    $daysStart = strtotime(date('d-m-Y', $startDate) . ' 00:00:00 tomorrow', $startDate);
+                    $this->SendDebug('StartDays', date('d.m.Y H:i:s', $daysStart), 0);
+                    $daysEnd = strtotime(date('d-m-Y', $endDate) . ' yesterday 23:59:59', $endDate);
+                    $this->SendDebug('EndDays', date('d.m.Y H:i:s', $daysEnd), 0);
+                    $days = AC_GetAggregatedValues($acID, $variableID, 1 /* Day */, $daysStart, $daysEnd, 0);
+
+                    //LastHours
+                    $lastHoursStart = strtotime('00:00:00', $endDate);
+                    $this->SendDebug('LastHoursStart', date('d.m.Y H:i:s', $lastHoursStart), 0);
+                    $lastHoursEnd = strtotime(date('H:59:59', $endDate) . ' - 1 hour', $endDate);
+                    $this->SendDebug('LastHoursEnd', date('d.m.Y H:i:s', $lastHoursEnd), 0);
+                    $lastHours = AC_GetAggregatedValues($acID, $variableID, 0 /* Hour */, $lastHoursStart, $lastHoursEnd, 0);
```

Comment:
Der Block fliegt dir um die Ohren, wenn, wie schon in der letzten Besprechung erwähnt, das Startdatum = das Enddatum ist